### PR TITLE
Feature provider has no way to ensure that their public key and address is indeed specified correctly in task to compute

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://github.com/golemfactory/golem-messages/issues/244.

### CODE REVIEW

File: golem_messages/message/tasks.py [hunk 1]
```diff
@@ -235,6 +235,9 @@ class WantToComputeTask(ConcentEnabled, base.Message):
         'price',
         'concent_enabled',  # Provider notifies requestor
                             # about his concent status
+        'provider_public_key',  # key used for msg signing and encryption
+        'provider_ethereum_public_key',  # used for transactions on blockchain
+        'eth_account',
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply golem_messages/message/tasks.py]: This was already there, and the issue description doesn't mention doing anything with it. Should it be removed or moved too ? If yes, please update issue description for consistency.

Btw. if this should be removed, please note that ie. Concent is using this field in few places.

File: golem_messages/message/tasks.py [hunk 2]
```diff
@@ -416,6 +430,10 @@ class ReportComputedTask(TaskMessage):
         'options',
     ] + base.Message.__slots__
 
+    @property
+    def eth_account(self):
+        return self.task_to_compute.eth_account
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply golem_messages/message/tasks.py]: The issue description says:

> following that, ReportComputedTask should not by itself contain the eth_account field and it should, again, be just a reference to the property in the TTC (which points to included WTCT)

That is why it is there. If it changes somehow, please update the issue description.

File: golem_messages/message/tasks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiivan) [reply golem_messages/message/tasks.py]: Please move `provider_ethereum_address` property to `WantToComputeTask`. There is no need to include both `provider_ethereum_public_key` and `eth_account` in one message. `eth_account` can be computed from `provider_ethereum_public_key`.

[the remark this answers is not in the document]
Comment (rwrzesien) [reply golem_messages/message/tasks.py]: Done, I have also move few tests which test this field on `TaskToCompute` to `WantToComputeTaskTest` test case.

File: tests/message/test_messages.py [hunk 3]
```diff
@@ -51,6 +52,8 @@ def test_message_want_to_compute_task(self):
         max_memory_size = random.randint(1, 2**10)
         num_cores = random.randint(1, 2**5)
         concent_enabled = random.random() > 0.5
+        provider_public_key = cryptography.ECCx(None).raw_pubkey
```
Comment (shadeofblue) [inline tests/message/test_messages.py]: hmm.... why this particular value? the message normally contains:

```
encode_key_id(cryptography.ECCx(None).raw_pubkey))
```

so, a hex-encoded representation of the public key... 

so, if we want to stay true to the actually expected content of those fields, we should use the above for both `provider_public_key` and `provider_ethereum_public_key` ... and - as is the case here - if we don't care about the value being a representation of an actual public key, then we can use a random string for both ...

Comment (rwrzesien) [reply tests/message/test_messages.py]: @shadeofblue :+1: done, I think it is better to use real values when possible.

File: golem_messages/message/tasks.py [hunk 4]
```diff
@@ -267,11 +274,18 @@ def requestor_ethereum_address(self):
...
-        )
+    def provider_public_key(self):
+        return self.want_to_compute_task.provider_public_key
+
+    @property
+    def provider_ethereum_public_key(self):
+        return self.want_to_compute_task.provider_ethereum_public_key
+
+    @property
+    def eth_account(self):
```
Comment (jiivan) [inline golem_messages/message/tasks.py]: Why change property name from `provider_ethereum_address` to `eth_account`?

Comment (rwrzesien) [reply golem_messages/message/tasks.py]: :+1: done, you are right, it is actually unnecessary change.

File: golem_messages/factories/tasks.py [hunk 5]
```diff
@@ -63,19 +66,14 @@ class Meta:
...
     compute_task_def = factory.SubFactory(ComputeTaskDefFactory)
-    provider_public_key = factory.LazyFunction(
-        lambda: encode_key_id(cryptography.ECCx(None).raw_pubkey))
-    provider_ethereum_public_key = factory.SelfAttribute(
-        'provider_public_key'
-    )
     requestor_public_key = factory.LazyFunction(
         lambda: encode_key_id(cryptography.ECCx(None).raw_pubkey))
 
-    compute_task_def = factory.SubFactory(ComputeTaskDefFactory)
```
Comment (jiivan) [inline golem_messages/factories/tasks.py]: :+1:

**Pull request merged**